src/components/avatar/index.tsx

```diff
@@ -384,7 +389,8 @@ class Avatar extends PureComponent<AvatarProps> {
         </View>
         {this.renderImage()}
         {this.renderBadge()}
-        {this.renderRibbon()}
+        {hasCustomRibbon && this.renderCustomRibbon()}
+        {!hasCustomRibbon && hasRibbonLabel && this.renderRibbon()}
```

Comment:
I'd remove here only the condition of `!hasCustomRibbon` 
Basically it means it will move the responsibility to the user. 
If the user pass both ribbonLabel and customRibbon they will have a UI bug and it should be their responsibility to handle it.